dojson/contrib/marc21/utils.py

```diff
@@ -30,48 +30,30 @@ def __new__(cls, values=None):
         new = dict.__new__(cls)
         dict.__init__(new)
 
-        data = []
-        keys = []
-        ordering = []
-
         if values:
-            if isinstance(values, GroupableOrderedDict):
-                values = values.items(repeated=True)
             if isinstance(values, dict):
-                if "__order__" not in values:
-                    raise AttributeError("Can only be recreated from an "
-                                         "ordered dict.")
-
-                ordering = values["__order__"]
-                occurences = Counter(ordering)
-                counter = Counter()
-                for key in ordering:
-                    if key not in keys:
-                        keys.append(key)
-                    if occurences[key] > 1:
-                        value = values[key][counter[key]]
-                    else:
-                        value = values[key]
-                    data.append((key, value))
-                    counter[key] += 1
+                new = dict.copy(values)
             else:
+                tmp = {}
+                ordering = []
+
                 for key, value in values:
-                    if key not in keys:
-                        keys.append(key)
-                    data.append((key, value))
+                    if key not in tmp:
```

Comment:
after https://github.com/inveniosoftware/dojson/pull/56/files#r47129235 it should not be necessary